plugins/repository-gcs/src/main/java/io/crate/gcs/GCSBlobContainer.java

```diff
@@ -21,165 +21,85 @@
-                return Channels.newInputStream(channel);
-            } else {
-                throw new IOException("blob `" + blobName + "` does not exist");
-            }
-        }
+    public Map<String, BlobMetadata> listBlobsByPrefix(String prefix) throws IOException {
+        return blobStore.listBlobsByPrefix(path, prefix);
     }
 
     @Override
-    public void writeBlob(String blobName, InputStream inputStream,
-                          long blobSize, boolean failIfAlreadyExists) throws IOException {
-        Blob blob = createBlob(blobName, failIfAlreadyExists);
-        try (WriteChannel channel = blob.writer()) {
-            final OutputStream outputStream = Channels.newOutputStream(channel);
-            inputStream.transferTo(outputStream);
-        }
+    public InputStream readBlob(String blobName) throws IOException {
+        return blobStore.readBlob(buildKey(blobName));
     }
 
     @Override
-    public void delete() {
-        String basePath = path().buildAsString();
-        var blobs = blobStore.storage().list(blobStore.bucketName(), prefix(basePath));
-        for (Blob blob : blobs.iterateAll()) {
-            blob.delete();
-        }
+    public InputStream readBlob(final String blobName, final long position, final long length) throws IOException {
+        return blobStore.readBlob(buildKey(blobName), position, length);
     }
 
     @Override
-    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) {
-        for (var blobName : blobNames) {
-            final Blob blob = getBlob(blobName);
-            if (blob != null) {
-                blob.delete();
-            }
-        }
+    public void writeBlob(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException {
+        blobStore.writeBlob(buildKey(blobName), inputStream, blobSize, failIfAlreadyExists);
     }
 
     @Override
-    public Map<String, BlobContainer> children() throws IOException {
-        String prefix = path().buildAsString();
-        var result = new HashMap<String, BlobContainer>();
-        var blobs = blobStore.storage().list(blobStore.bucketName(), currentDirectory(), prefix(prefix));
-        for (var blob : blobs.iterateAll()) {
-            if (blob.isDirectory()) {
-                assert blob.getName().startsWith(prefix);
-                assert blob.getName().endsWith("/");
-                final String suffixName = blob.getName().substring(prefix.length(), blob.getName().length() - 1);
-                if (suffixName.isEmpty() == false) {
-                    result.put(suffixName, new GCSBlobContainer(path().add(suffixName), blobStore));
-                }
-            }
-        }
-        return result;
+    public void writeBlobAtomic(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException {
+        writeBlob(blobName, inputStream, blobSize, failIfAlreadyExists);
```

Comment:
is this correct? the method is `writeBlobAtomic` but calls the above method, and not `blobStore.writeBlob`, I know it's the same as in original code, just double checking if something here is missing.